src/utils/staking.js

```diff
@@ -0,0 +1,278 @@
+        )).filter((v) => !!v)
+    }
+
+    // useLockup will be set by user in redux state and passed through actions
+    async stake(useLockup, validatorId, amount) {
+        amount = parseNearAmount(amount)
+        const { contract, lockupId } = await this.getLockup()
+        const selectedValidatorId = await contract.get_staking_pool_account_id()
+        console.log('selectedValidatorId', selectedValidatorId)
+        console.log('validatorId', validatorId)
+        if (validatorId !== selectedValidatorId) {
+            await this.lockupSelect(validatorId, lockupId, selectedValidatorId !== null)
+        }
+        return await this.lockupStake(lockupId, validatorId, amount)
+    }
+
+    async unstake(useLockup) {
+        const { lockupId } = await this.getLockup()
+        const unstake_all = await this.signAndSendTransaction(lockupId, [
+            functionCall('unstake_all', {}, GAS_STAKE)
+        ])
+        console.log('unstake_all', unstake_all)
+    }
+
+    async withdraw(useLockup) {
+        const { lockupId } = await this.getLockup()
+        const withdraw_all_from_staking_pool = await this.signAndSendTransaction(lockupId, [
+            functionCall('withdraw_all_from_staking_pool', {}, GAS_STAKE)
+        ])
+        console.log('withdraw_all_from_staking_pool', withdraw_all_from_staking_pool)
+        if (withdraw_all_from_staking_pool[0] === false) {
+            throw new Error('Unable to withdraw pending balance from validator')
+        }
+    }
+
+    // helpers for lockup
+
+    async lockupStake(lockupId, validatorId, amount) {
+        const deposit_and_stake = await this.signAndSendTransaction(lockupId, [
+            functionCall('deposit_and_stake', { amount }, GAS_STAKE)
+        ])
+        console.log('deposit_and_stake', deposit_and_stake)
+        return deposit_and_stake
+    }
+
+    async lockupSelect(validatorId, lockupId, unselect = false) {
+        if (unselect) {
+            try {
+                const unselect_staking_pool = await this.signAndSendTransaction(lockupId, [
+                    functionCall('unselect_staking_pool', {}, GAS_STAKE)
+                ])
+                console.log('unselect_staking_pool', unselect_staking_pool)
+            } catch (e) {
+                console.warn('Problem unselecting validator', validatorId, e)
+            }
+        }
+        try {
+            const select_staking_pool = await this.signAndSendTransaction(lockupId, [
+                functionCall('select_staking_pool', { staking_pool_account_id: validatorId }, GAS_STAKE)
+            ])
+            console.log('select_staking_pool', select_staking_pool)
+        } catch (e) {
+            console.warn('Problem selecting validator', validatorId, e)
+            throw new Error('Cannot select validator')
+        }
+    }
+
+    async getLockup() {
+        const accountId = this.wallet.accountId
+
+        let lockupId
+        if (process.env.REACT_APP_NETWORK_ID === 'mainnet') {
```

Comment:
let's not hardcode anything for specific networks. If something needs to be overriden – we can use more specific env vars to do it (like e.g. path lockup account name pattern, etc)